File: 12-integer-to-roman/integer-to-roman.cs
public class Solution {
    public string IntToRoman(int num) {
        string[] thousands = { "", "M", "MM", "MMM" };
        string[] hundreds = { "", "C", "CC", "CCC", "CD", "D", "DC", "DCC", "DCCC", "CM" };
        string[] tens = { "", "X", "XX", "XXX", "XL", "L", "LX", "LXX", "LXXX", "XC" };
        string[] ones = { "", "I", "II", "III", "IV", "V", "VI", "VII", "VIII", "IX" };
        
        string roman = thousands[num / 1000] + 
                       hundreds[(num % 1000) / 100] + 
                       tens[(num % 100) / 10] + 
                       ones[num % 10];
        
        return roman;
    }
}

File: 12-integer-to-roman/integer-to-roman.js
/**
 * @param {number} num
 * @return {string}
 */
var intToRoman = function(num) {
    const romanSymbols = [
        ['M', 1000],
        ['CM', 900],
        ['D', 500],
        ['CD', 400],
        ['C', 100],
        ['XC', 90],
        ['L', 50],
        ['XL', 40],
        ['X', 10],
        ['IX', 9],
        ['V', 5],
        ['IV', 4],
        ['I', 1]
    ];

    let roman = '';
    for (const [symbol, value] of romanSymbols) {
        while (num >= value) {
            roman += symbol;
            num -= value;
        }
    }
    return roman;
};
